components/merkledb/src/group.rs

```diff
@@ -51,6 +52,7 @@ where
     I: FromAccess<T>,
 {
     fn from_access(access: T, addr: IndexAddress) -> Result<Self, AccessError> {
+        assert_valid_index_name_component(&addr.name);
```

Comment:
Why is naming checked for component here? It should be possible to create a group with a name like `foo.bar`, isn't it?